### PR TITLE
[example] update apiVersion of Deployment

### LINKED ISSUE #3160: 2 examples are incompatible with k8s v1.16.
### Expected behavior

Examples are successfully deployed to k8s v1.16 cluster.

- helm-deployment
- helm-deployment-dependencies

### Actual behavior

`skaffold dev` exits with an error.

```
$ cd ~/go/src/github.com/GoogleContainerTools/skaffold/examples/helm-deployment
$ skaffold dev
Listing files to watch...
 - gcr.io/decent-space-249411/skaffold-helm
Generating tags...
 - gcr.io/decent-space-249411/skaffold-helm -> gcr.io/decent-space-249411/skaffold-helm:latest
Checking cache...
 - gcr.io/decent-space-249411/skaffold-helm: Found Locally
Tags used in deployment:
 - gcr.io/decent-space-249411/skaffold-helm -> gcr.io/decent-space-249411/skaffold-helm:3987783aa1d45ad1c1ea91d31f52d51549db25986c58676ae1141f4ccd547b4c
   local images can't be referenced by digest. They are tagged and referenced by a unique ID instead
Starting deploy...
Helm release skaffold-helm not installed. Installing...
No requirements found in skaffold-helm/charts.
Error: validation failed: unable to recognize "": no matches for kind "Deployment" in version "extensions/v1beta1"
WARN[0001] error retrieving helm deployment info: Error: release: "skaffold-helm" not found 
Cleaning up...
Error: release: "skaffold-helm" not found
FATA[0001] exiting dev mode because first deploy failed: deploying skaffold-helm: exit status 1 
```

### Information

- Skaffold version: v0.41.0
- Operating system: macOS Catalina 10.15.1
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta17
kind: Config
build:
  tagPolicy:
    sha256: {}
  artifacts:
  - image: gcr.io/decent-space-249411/skaffold-helm
deploy:
  helm:
    releases:
    - name: skaffold-helm
      chartPath: skaffold-helm
      #wait: true
      #valuesFiles:
      #- helm-skaffold-values.yaml
      values:
        image: gcr.io/decent-space-249411/skaffold-helm
      #recreatePods will pass --recreate-pods to helm upgrade
      #recreatePods: true
      #overrides builds an override values.yaml file to run with the helm deploy
      #overrides:
      # some:
      #   key: someValue
      #setValues get appended to the helm deploy with --set.
      #setValues:
        #some.key: someValue
```
https://github.com/GoogleContainerTools/skaffold/blob/master/examples/helm-deployment/skaffold.yaml

### Steps to reproduce the behavior

1. `minikube start --kubernetes-version=v1.16.2`
2. `kubectl apply -f ~/go/src/gitlab.com/cloudnative_impress/tutorial/chap06/helm/tiller-rbac.yaml`
3. `helm init --service-account tiller`
4. `git clone https://github.com/GoogleContainerTools/skaffold.git`
5. `cd skaffold/examples/helm-deployment`
6. `skaffold dev`

Deployment apiVersion should be updated to apps/v1.

### PR DESCRIPTION
**Pull request by oke-py**
Relates to _in case of new feature, this should point to issue/(s) which describes the feature_

Fixes #3160.  _in case of a bug fix, this should point to a bug and any other related issue(s)_

Should merge before : _list any PRs that depend on this PR_

Should merge after : _list any PRs that are prerequisites to this PR_

**Description**

**User facing changes**

n/a

**Next PRs.**

n/a

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [X] Mentions any output changes.
- [X] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [X] Adds integration tests if needed.

**Reviewer Notes**

- [x] The code flow looks good. 
- [x] Unit test added.
- [x] User facing changes look good.

**Release Notes**

```
Update examples apiVersion of Deployment, Ingress
```

### CODE REVIEW

File: examples/kustomize/skaffold-kustomize-args.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: skaffold/v1beta16
+apiVersion: skaffold/v1beta17
```
Comment (dgageot) [inline examples/kustomize/skaffold-kustomize-args.yaml]: please rebase your PR. this should already be fixed

File: examples/profile-patches/skaffold.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-apiVersion: skaffold/v1beta16
+apiVersion: skaffold/v1beta17
```
Comment (dgageot) [inline examples/profile-patches/skaffold.yaml]: please rebase your PR. this should already be fixed

**Comment by dgageot** [comment]
Please rebase your PR and them it's all good!

**Comment by oke-py** [comment]
I rebased and force-pushed

**Comment by dgageot** [comment]
@oke-py you rock!

**Comment by oke-py** [comment]
Travis CI failed because of master branch. I rebased again to take in the patch for master.
Need kokoro:run?

**Pull request merged**